layout_padded/layout_padded.md

```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
 
 ## Layout specializations of `submdspan_mapping` [mdspan.submdspan.mapping]
 
-> At the top of Section � [mdspan.submdspan.mapping] ("Layout specializations of `submdspan_mapping`"), before paragraph 1, add the following to the end of the synopsis of specializations.
+> Replace Section � [mdspan.submdspan.mapping] ("Layout specializations of `submdspan_mapping`"), with:
 
-```c++
-  template<class Extents, std::size_t padding_stride, class... SliceSpecifiers>
-    constexpr auto submdspan_mapping(
-      const layout_left_padded<padding_stride>::template mapping<Extents>& src, 
-      SliceSpecifiers ... slices) -> /* see below */;
+<b>24.7.3.7.6 Specialization of `submdspan_mapping` [mdspan.submdspan.mapping]</b>
+
+<b>24.7.3.7.6.1 Common [mdspan.submdspan.mapping.common]</b>
+
+[1]{.pnum} The following elements apply to all functions in [mdspan.submdspan.impl].
+
+[2]{.pnum} *Constraints:* `sizeof...(slices)` equals `extents_type::rank()`,
+
+[3]{.pnum} *Mandates:* For each rank index `k` of `extents()`, exactly one of the following is true:
+
+   * [3.1]{.pnum} $S_k$ models `convertible_to<index_type>`
+
+   * [3.2]{.pnum} $S_k$ models _`index-pair-like`_`<index_type>`
+
+   * [3.3]{.pnum} `is_convertible_v<`$S_k$`, full_extent_t>` is `true`
+
+   * [3.4]{.pnum} $S_k$ is a specialization of `strided_slice`
+
+[4]{.pnum} *Preconditions:* For each rank index `k` of `extents()`, all of the following are true:
+
+   * [4.1]{.pnum}  if $S_k$ is a specialization of `strided_slice`
+
+      * $s_k$`.extent` $= 0$, or
+
+      * $s_k$`.stride` $\gt 0$
+
+   * [4.2]{.pnum} $0 \le$ _`first`_`_<index_type, k>(slices...)` $\le$ _`last_<k>`_`(extents(), slices...)` $\le$ `extents.extent(k)`
 
-  template<class Extents, std::size_t padding_stride, class... SliceSpecifiers>
-    constexpr auto submdspan_mapping(
-      const layout_right_padded<padding_stride>::template mapping<Extents>& src, 
-      SliceSpecifiers ... slices) -> /* see below */;
+[5]{.pnum} Let `sub_ext` be the result of `submdspan_extents(extents(), slices...)` and let `SubExtents` be `decltype(sub_ext)`.
+
+[6]{.pnum} Let `sub_strides` be an `array<SubExtents::index_type, SubExtents::rank()>` such that for each rank index `k` of `extents()` for which _`map-rank`_`[k]` is not `dynamic_extent`, `sub_strides[`_`map-rank`_`[k]]` equals:
+
+   * [6.1]{.pnum} `stride(k) * `_`de-ice`_`(`$s_k$`.stride)` if $S_k$ is a specialization of `strided_slice` and $s_k$`.stride` $\lt s_k$`.extent`;
+
+   * [6.2]{.pnum} otherwise, `stride(k)`.
+
+[7]{.pnum} Let `P`  be a parameter pack such that `is_same_v<make_index_sequence<rank()>, index_sequence<P...>>` is `true`.
+
+[8]{.pnum} Let `offset` be a value of type `size_t` equal to `(*this)(`_`first`_`_<index_type, P>(slices...)...)`.
+
```

Comment:
Everything after this point diverges from the current Standard (the online draft includes `submdspan`; see https://eel.is/c++draft/views.multidim#mdspan.submdspan.submdspan ).

Reply:
yeah I know: because I rewrote it all